azure-kusto-ingest/azure/kusto/ingest/ingestion_properties.py

```diff
@@ -69,15 +71,75 @@ class ReportMethod(IntEnum):
     Queue = 0
 
 
+class TransformationMethod(Enum):
+    # Transformations to configure over json column mapping
+    NONE = "None"
+    PROPERTY_BAG_ARRAY_TO_DICTIONARY = "PropertyBagArrayToDictionary",
+    SOURCE_LOCATION = "SourceLocation"
+    SOURCE_LINE_NUMBER = "SourceLineNumber"
+    GET_PATH_ELEMENT = "GetPathElement"
+    UNKNOWN_ERROR = "UnknownMethod"
+    DATE_TIME_FROM_UNIX_SECONDS = "DateTimeFromUnixSeconds"
+    DATE_TIME_FROM_UNIX_MILLISECONDS = "DateTimeFromUnixMilliseconds"
+    DATE_TIME_FROM_UNIX_MICROSECONDS = "DateTimeFromUnixMicroseconds"
+    DATE_TIME_FROM_UNIX_NANOSECONDS = "DateTimeFromUnixNanoseconds"
+
+
 class ColumnMapping:
-    """Abstract class to column mapping."""
+    """Common class to all the column mapping kinds."""
+    """Mapping properties keys."""
+    # Json Mapping consts
+    PATH = "Path",
+    TRANSFORMATION_METHOD = "Transform"
+    # csv Mapping consts
+    ORDINAL = "Ordinal"
+    CONST_VALUE = "ConstValue"
+    # Avro Mapping consts
+    FIELD_NAME = "Field"
+    COLUMNS = "Columns"
+    # General Mapping consts
+    STORAGE_DATA_TYPE = "StorageDataType"
 
+    def __init__(
+        self,
+        column_name,
+        column_type,
+        path=None,
+        transform=TransformationMethod.NONE,
+        ordinal=None,
+        const_value=None,
+        field=None,
+        columns=None,
+        storage_data_type=None,
+    ):
+        self.column = column_name
+        self.datatype = column_type
+        self.properties = {}
+        if path:
+            self.properties[self.PATH] = path
```

Comment:
i'm not sure I under stand why not just do:
`self.path` instead of `self.properties.path`, but I'm assuming it's related to the way we stringify this,
so I would perhaps go with adding setters/ getter for easier usage (in python that would be using `@property` and `@propery.setter`

Reply:
i was asked by @ariel yehezkely not to use setters and getters as its not common in python and in this sdk .. i think he is right about that

Reply:
and yes- i do it for the json created ..